emop-maven-plugin/src/main/java/edu/cornell/RppHandlerMojo.java

```diff
@@ -0,0 +1,167 @@
+    private void computeSpecSets() {
+        Set<String> allSpecs = Util.retrieveSpecListFromJar(javamopAgent);
+        // if we still don't have any spec files, then we'd just need to obtain all specs and run it in critical
+        if (criticalSpecsFile == null && backgroundSpecsFile == null) {
+            criticalSpecsSet = allSpecs;
+            backgroundSpecsSet = new HashSet<>();
+        } else if (backgroundSpecsFile == null) {
+            // if we only have the critical specs, then our background specs is the set difference
+            criticalSpecsSet = parseSpecsFile(criticalSpecsFile);
+            backgroundSpecsSet = Sets.difference(allSpecs, criticalSpecsSet);
+        } else if (criticalSpecsFile == null) {
+            // if we only have the background specs, then our critical specs is the set difference
+            backgroundSpecsSet = parseSpecsFile(backgroundSpecsFile);
+            criticalSpecsSet = Sets.difference(allSpecs, backgroundSpecsSet);
+        } else {
+            // we have both files, so all we need to do is read from them.
+            criticalSpecsSet = parseSpecsFile(criticalSpecsFile);
+            backgroundSpecsSet = parseSpecsFile(backgroundSpecsFile);
+            backgroundSpecsSet.removeAll(criticalSpecsSet);
+            if (criticalSpecsSet.isEmpty() && backgroundSpecsSet.isEmpty()) {
+                getLog().error("Both critical sets and background sets were empty!");
+                System.exit(1);
+            }
+        }
+    }
+
+    /**
+     * Creates a JavaMOP agent JAR configured to only monitor the specified set of specifications.
+     * @param mode an identifier for the jar (either "critical" or "background").
+     * @param specsToMonitor the list of specifications the agent should record.
+     * @return The path to the created JAR.
+     */
+    private String setUpSingleJar(String mode, Set<String> specsToMonitor) {
+        if (specsToMonitor.isEmpty()) {
+            return "";
+        } else {
+            File javamopAgentFile = new File(javamopAgent);
+            File createdJar = new File(metaInfoDirectory, mode + "-javamop.jar");
+            try {
+                Files.copy(javamopAgentFile.toPath(), createdJar.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                Util.generateNewMonitorFile(
+                        metaInfoDirectory + File.separator + mode + "-ajc.xml", specsToMonitor);
+                Util.replaceFileInJar(createdJar.getAbsolutePath(), "/META-INF/aop-ajc.xml",
+                        metaInfoDirectory + File.separator + mode + "-ajc.xml");
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            } catch (MojoExecutionException ex) {
+                throw new RuntimeException(ex);
+            }
+            return createdJar.getAbsolutePath();
+        }
+    }
+
+    /**
+     * Creates new agent JARs for running critical and background phases, and sets up System properties to record
```

Comment:
Do you enforce a line length in your checkstyle config? I usually enforce 120 characters per line

Reply:
There's an enforced line length but it's 125.